src/transformers/integrations/peft.py

```diff
@@ -15,6 +15,7 @@
 import inspect
 import warnings
 from typing import Any, Dict, List, Optional, Union
+from peft.tuners.tuners_utils import BaseTunerLayer
```

Comment:
Let's add a version check like over here:
https://github.com/huggingface/transformers/blob/c07db46abc37f0af1d8ebaba26641da90cf8c2d9/src/transformers/integrations/peft.py#L413-L421

And import it afterwards within the method, should fix most tests that are currently failing.

Reply:
Yes indeed